e2e/imgbuild/imgbuild.go

```diff
@@ -2033,6 +2033,15 @@ func (c *imgBuildTests) testContainerBuildUnderFakerootModes(t *testing.T) {
 		e2e.WithArgs("--force", "--ignore-userns", "--ignore-subuid", fmt.Sprintf("%s/openssh-mode4.sif", tmpDir), "testdata/unprivileged_build_4.def"),
 		e2e.ExpectExit(0),
 	)
+
+	// running under the fakeroot with unprivileged namespace unavailable
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.FakerootProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--force", "--ignore-userns", fmt.Sprintf("%s/fakeroot-no-userns-build.sif", tmpDir), "testdata/fakeroot_no_userns_build.def"),
+		e2e.ExpectExit(0),
+	)
 }
```

Comment:
No, this test is essentially the same as the previous one for mode 4.

No I think the problem with the tests in this section is that they should all be run with UserProfile instead of UsernamespaceProfile or FakerootProfile.  I see that the build works if I add a `--fakeroot` option, but that is suppposed to be implied.

Reply:
I updated the PR for this.

Reply:
I verified that even with unprivileged user namespaces enabled, the `--ignore-userns` option without an expicit `--fakeroot` exhibited the same symptom before this PR.  By using the FakerootProfile in the mode 4 e2e test, the `--fakeroot` option was passed and that hid the error.

Reply:
@DrDaveD thanks so much for the updates. ahh, I get the point, actually when we are using `FakerootProfile`, it equals to adding an `--fakeroot` option to apptainer. so we do not need duplicated declaration.